### ISSUE TITLE
Rule Change: max-statements; add ignore option

### ISSUE THREAD

**Issue by reddevil7nine** [eslint/eslint#18651]
### What rule do you want to change?

max-statements

### What change do you want to make?

Generate fewer warnings

### How do you think the change should be implemented?

A new option

### Example code

```js
// eslint.config.mjs
"max-statements": [
   "error",
   { max: 8 }
]

// code
function save()
{
   logInfo( "start" );
   
   const coolVariable = someFunction();
   let coolStuff = true;
   
   logInfo( coolStuff );

   if ( coolVariable === "a cool value" )
   {
      logInfo( "doing cool stuff" );
   
      doSomeMoreCoolStuff();
      doEventMoreCoolStuff();
   }
   else
   {
      logInfo( "doing not so cool stuff" );
   
      doSomeNotSoCoolStuff();
      coolStuff = false;
   }

   logInfo( "end" );
   
   return coolStuff;
}
```

### What does the rule currently do for this code?

Gives the following message:
`// error Method 'save' has too many statements (13). Maximum allowed is 8 max-statements`

`max-statements` produces an error because it's including the calls to `logInfo` in it's evaluation. While `logInfo` does add complexity to the function, a Developer reading the code would ignore the calls to `logInfo` since there would be an understanding that it doesn't affect how the function works.

### What will the rule do after it's changed?

Allowing a regex option or keywords to ignore certain code would allow `max-statements` to provide Developers a way to ignore code that is "unimportant" when evaluating the true complexity of the function. Otherwise, some "helpful" code that a Developer would want to add would have to be sacrificed in order for the rule to pass.

```
// eslint.config.mjs
"max-statements": [
   "error",
   { 
      max: 8,
      ignore: "^logInfo\\("
   }
]
```

### Participation

- [ ] I am willing to submit a pull request to implement this change.

**Comment by nzakas** [comment]
Thanks for the input. As [stated in the README](https://github.com/eslint/eslint#stylistic-rule-updates), we actually aren't making any further updates to rules that enforce stylistic preferences. If you have a strong preference for such a change, you can always copy the existing rule into a custom rule and make it behave the way you want.
